climpred/bootstrap.py

```diff
@@ -278,24 +281,43 @@ def bootstrap_compute(
     ci_high_pers = 1 - p_pers / 2
 
     inits = hind.init.values
+    members = hind.member.values
     init = []
     uninit = []
     pers = []
+
+    if dim == 'member':
+        to_be_shuffled = members
+        shuffle_dim = 'member'
+    elif 'init' in dim:  # also allows ['init','member']
+        to_be_shuffled = inits
+        shuffle_dim = 'init'
+    else:
+        raise ValueError('Shuffle either `member` or `init`; not', dim)
     # resample with replacement
     # DoTo: parallelize loop
     for _ in range(bootstrap):
-        smp = np.random.choice(inits, len(inits))
-        smp_hind = hind.sel(init=smp)
+        smp = np.random.choice(to_be_shuffled, len(to_be_shuffled))
+        smp_hind = hind.sel({shuffle_dim: smp})
+        if shuffle_dim == 'member':
+            smp_hind['member'] = np.arange(1, 1 + smp_hind.member.size)
         # compute init skill
-        init.append(
-            compute(
-                smp_hind,
-                reference,
-                metric=metric,
-                comparison=comparison,
-                add_attrs=False,
-            )
+        init_skill = compute(
+            smp_hind,
+            reference,
+            metric=metric,
+            comparison=comparison,
+            add_attrs=False,
+            dim=dim,
+            **kwargs,
         )
+        if (
```

Comment:
Shouldn't there be an `else` here to throw some error or something? Currently this is checking 3 conditions, so there's a lot of alternatives here. Maybe it doesn't matter.

Reply:
This is more correction function. Only in these conditions, I need to reset init labels. Therefore no else